lsp/autotools_ls.lua

```diff
@@ -15,8 +13,12 @@ local root_files = { 'configure.ac', 'Makefile', 'Makefile.am', '*.mk' }
 return {
   cmd = { 'autotools-language-server' },
   filetypes = { 'config', 'automake', 'make' },
-  root_dir = function(bufnr, on_dir)
-    local fname = vim.api.nvim_buf_get_name(bufnr)
-    on_dir(util.root_pattern(unpack(root_files))(fname))
+  root_markers = function(name, _)
+    for _, pattern in ipairs(root_files) do
+      if vim.glob.to_lpeg(pattern):match(name) ~= nil then
```

Comment:
If this is the pattern that's actually unavoidable in every config, it's a strong hint that we need to resolve https://github.com/neovim/neovim/issues/33444 (by enhancing `vim.fs.find()`, and/or accepting a function for `root_markers`, and/or supporting wildcards directly in `root_dir` / `root_markers`).